### PR TITLE
oneMKL: support for swap

### PR DESCRIPTION
**Pull request by Sarbojit2019**

### CODE REVIEW

**Comment by Sarbojit2019** [comment]
Failure does not look related hence closing and re-opening to restart the build.

**Comment by Sarbojit2019** [comment]
> The MKL wrapper for `copy` seems broken, returning all-zeros. Although not strictly related to this PR, it's related to recent oneMKL work. Could you take a look?

Sure! let me check if I can reproduce the issue.

**Comment by Sarbojit2019** [comment]
@maleadt, Build is clean now, can you please take a look, if it is good to go?

File: test/onemkl.jl [hunk 1]
```diff
@@ -16,14 +16,27 @@ m = 20
...
+        @testset "swap" begin
+            # testing swap
+            x = rand(T, m)
+            y = rand(T, m)
+            dx = oneArray(x)
+            dy = oneArray(y)
+            oneMKL.swap!(m, dx, dy)
+            h_x = collect(dx)
+            h_y = collect(dy)
+            @test h_x ≈ y
```
[the remark this answers is not in the document]
Comment (maleadt) [reply test/onemkl.jl:38]: I pushed a commit to fix this.

**Pull request merged**